packages/visualizations/src/components/Map/MapRender.svelte

```diff
@@ -50,6 +58,22 @@ TODO:
             ...start,
         });
 
+        // Set a resizeObserver to resize map on container size changes
+        resizer = new ResizeObserver(debounce(() => {
+            map.resize();
+            // Cancel saved max bounds to properly fitBounds
+            map.setMaxBounds(null);
+            if (bbox) {
```

Comment:
I'm not sure in which situations we don't have a bbox yet, but reading the code naïvely, I suppose the two calls to setMaxBounds could be inside the `if` clause, since they are only useful if we do `fitBounds`?

Also, once this is done, I guess theses three lines could be in a function, because the exact same lines are used in two places. Something like

```js
function fitMapToBbox(newBbox) {
    // Cancel saved max bounds to properly fitBounds
    map.setMaxBounds(null);
    map.fitBounds(newBbox, {
        animate: false,
        padding: 10,
    });
    // Rest min zoom and movement
    map.setMaxBounds(map.getBounds());
}
```

Reply:
Good remark ! I've moved the three lines into one function. As for existence check on bbox, I noticed that on map initialization the ResizeObserver can be called and then bbox is undefined.